### PR TITLE
feat: add filters for expenses

### PR DESCRIPTION
**Pull request by LizavetaY**
### Description

Added a filters component for page 'Expenses' and themes for styling List and calendar elements. Added libraries for work with dates, Selects and calendar

### Type of change

- [x] Feature (introduces new functionality)
- [ ] Bugfix
- [ ] Configuration
- [ ] Refactoring (doesn't change functionality, only the code)
- [ ] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)
- [x] Chakra components/constants/hooks have been used wherever possible

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

Add screenshots from the [current deployed state](https://sandbox-money-app.herokuapp.com/) of the changed components here.

#### Screenshots/Video after change

Снимок экрана 2022-07-29 в 1 11 40 PM
Снимок экрана 2022-07-29 в 1 11 51 PM
Снимок экрана 2022-07-29 в 1 12 10 PM

https://user-images.githubusercontent.com/47990337/181748564-4f0f79ae-b5f8-4f3c-b868-a44d5af744ab.mov

https://user-images.githubusercontent.com/47990337/181748607-8104cca2-a2cf-4873-ae8a-f1f16f505a98.mov

https://user-images.githubusercontent.com/47990337/181748878-a670e5ca-eeac-4b33-8e9d-4d1a018c16d5.mov

**Commit 'fix: add logic for picked days'**
Logic: if the user selects the range and it matches the days from the suggested days' list, the correct range on the list will be checked

https://user-images.githubusercontent.com/47990337/182032335-f22e5028-2420-44de-8d7d-49c55d3c08a4.mov

### CODE REVIEW

File: src/components/FiltersExpenses/CalendarPicker.jsx [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+      setIsCustomizedRange(datesObj.lastWeek);
+    } else if (isDateButtonSelected.thisMonth) {
+      setIsCustomizedRange(datesObj.thisMonth);
+    } else if (isDateButtonSelected.lastMonth) {
+      setIsCustomizedRange(datesObj.lastMonth);
+    } else if (!!chosenDates.start || !!chosenDates.end) {
+      setIsCustomizedRange({ start: null, end: null });
+    }
+  }, [chosenDates]);
+
```
Comment (PavelTitov16) [inline src/components/FiltersExpenses/CalendarPicker.jsx:232]: maybe better to use switch / case here?

Comment (PavelTitov16) [inline src/components/FiltersExpenses/CalendarPicker.jsx:232]: maybe better to use switch / case here?

File: src/components/FiltersExpenses/CalendarPicker.jsx [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    } else if (isDateButtonSelected.lastWeek) {
+      setIsCustomizedRange(datesObj.lastWeek);
+    } else if (isDateButtonSelected.thisMonth) {
+      setIsCustomizedRange(datesObj.thisMonth);
+    } else if (isDateButtonSelected.lastMonth) {
+      setIsCustomizedRange(datesObj.lastMonth);
+    } else if (!!chosenDates.start || !!chosenDates.end) {
+      setIsCustomizedRange({ start: null, end: null });
+    }
+  }, [chosenDates]);
```
Comment (PavelTitov16) [inline src/components/FiltersExpenses/CalendarPicker.jsx:231]: maybe better to use switch / case here?

File: src/components/FiltersExpenses/CalendarPicker.jsx [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+      setIsCustomizedRange(datesObj.thisWeek);
+    } else if (isDateButtonSelected.lastWeek) {
+      setIsCustomizedRange(datesObj.lastWeek);
+    } else if (isDateButtonSelected.thisMonth) {
+      setIsCustomizedRange(datesObj.thisMonth);
+    } else if (isDateButtonSelected.lastMonth) {
+      setIsCustomizedRange(datesObj.lastMonth);
+    } else if (!!chosenDates.start || !!chosenDates.end) {
+      setIsCustomizedRange({ start: null, end: null });
+    }
```
Comment (PavelTitov16) [inline src/components/FiltersExpenses/CalendarPicker.jsx:230]: maybe better to use switch / case here?

Comment (PavelTitov16) [inline src/components/FiltersExpenses/CalendarPicker.jsx:230]: maybe better to use switch / case here?

Comment (LizavetaY) [reply src/components/FiltersExpenses/CalendarPicker.jsx:176]: Change
Снимок экрана 2022-07-30 в 4 04 00 PM

**Review by itsnotpolza — APPROVED** [review]

File: src/components/FiltersExpenses/CalendarPicker.jsx [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+              <CalendarDays />
+            </CalendarMonth>
+          </CalendarMonths>
+
+          <Text ml="10px">
+            {i18next.t('expenses.filters.date.calendar.range.name')}
+            {daysNum}{' '}
+            {daysNum === 1
+              ? i18next.t('expenses.filters.date.calendar.range.day')
+              : i18next.t('expenses.filters.date.calendar.range.days')}
```
Comment (kalashnikau) [inline src/components/FiltersExpenses/CalendarPicker.jsx:272]:
```suggestion
            {i18n.t('expenses.filters.date.calendar.range.day', { count: daysNum })}
```

Comment (kalashnikau) [reply src/components/FiltersExpenses/CalendarPicker.jsx:272]: and then in the language file

`"expenses.filters.date.calendar.range.day": {
    "one": "day",
    "other": "days"
  },`

File: src/components/FiltersExpenses/CalendarPicker.jsx [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    });
+
+    setIsCheckedIcon('thisMonth');
+  };
+
+  const setLastMonth = (event) => {
+    setChosenDates(datesObj.lastMonth);
+
+    setSelectedDateFilter({
+      ...selectedDateFilter,
```
[the remark this answers is not in the document]
Comment (LizavetaY) [reply src/components/FiltersExpenses/CalendarPicker.jsx:141]: Снимок экрана 2022-08-01 в 11 17 18 AM

File: src/components/FiltersExpenses/CalendarPicker.jsx [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+        break;
+      case !!chosenDates.start || !!chosenDates.end:
+        changeToRightRange({ start: null, end: null });
+        break;
+    }
+  }, [chosenDates]);
+
+  return (
+    <Flex alignItems="stretch" rounded="md" overflow="hidden">
+      <List p="20px 0" w="200px">
```
[the remark this answers is not in the document]
Comment (LizavetaY) [reply src/components/FiltersExpenses/CalendarPicker.jsx:199]: Снимок экрана 2022-08-01 в 11 17 56 AM

File: src/components/FiltersExpenses/CalendarPicker.jsx [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+    } else {
+      setDaysNum(
+        differenceInCalendarDays(chosenDates.end, chosenDates.start) + 1 || 0
+      );
+    }
+
+    setSelectedDateFilter({ ...selectedDateFilter, dates: chosenDates });
+
+    switch (true) {
+      case isDateButtonSelected.today:
```
[the remark this answers is not in the document]
Comment (LizavetaY) [reply src/components/FiltersExpenses/CalendarPicker.jsx:173]: Снимок экрана 2022-08-01 в 11 21 25 AM

File: src/components/FiltersExpenses/CalendarPicker.jsx (hunk 4 above)
Comment (LizavetaY) [reply src/components/FiltersExpenses/CalendarPicker.jsx:272]: Снимок экрана 2022-08-01 в 11 21 58 AM
Снимок экрана 2022-08-01 в 11 22 10 AM

**Review by kalashnikau — APPROVED** [review]

**Review by PavelTitov16 — APPROVED** [review]

**Pull request merged**